drivers/flash/flash_stm32.h

```diff
@@ -54,18 +54,31 @@ struct flash_stm32_priv {
 
 /* Redefinitions of flags and masks to harmonize stm32 series: */
 #if defined(CONFIG_SOC_SERIES_STM32U5X)
-#define FLASH_NSCR_NSLOCK FLASH_NSCR_LOCK
-#define FLASH_OPTR_DBANK FLASH_OPTR_DUALBANK
-#define FLASH_NSCR_NSPG FLASH_NSCR_PG
-#define FLASH_NSCR_NSBKER_Msk FLASH_NSCR_BKER_Msk
```

Comment:
This whole part should ideally be moved into initial commit "drivers: flash stm32 flash driver includes stm32u5 serie".
But I agree this require a full rework of the PR, so ok to keep it that way.